components/src/dynamo/vllm/args.py

```diff
@@ -154,9 +188,19 @@ def parse_args() -> Config:
         # This becomes an `Option` on the Rust side
         config.served_model_name = None
 
-    config.namespace = os.environ.get("DYN_NAMESPACE", "dynamo")
-    config.component = "prefill" if args.is_prefill_worker else "backend"
-    config.endpoint = "generate"
+    if args.endpoint:
```

Comment:
Can you add context to PR description on what scenario we are fixing and when it would come up?